pkgs/applications/editors/vscode/generic.nix

```diff
@@ -108,6 +111,30 @@ let
       )
     '';
 
+    # See https://github.com/NixOS/nixpkgs/issues/49643#issuecomment-873853897
+    postPatch = ''
+      # PATCHING RUN AS SUDO
+      # --------------------
+      # where to find the node.js librar .asar was described here https://github.com/NixOS/nixpkgs/issues/76526#issuecomment-569131432
+      packed="resources/app/node_modules.asar"
+      # we unpack directly into the same name without .asar ending,
+      # which is adapted from hacking-skype-tutorial https://www.codepicky.com/hacking-electron-restyle-skype/
+      unpacked="resources/app/node_modules"
+
+      ${nodePackages.asar}/bin/asar extract "$packed" "$unpacked"
+      # we change paths to pkexec and bash
+      # as described here https://github.com/NixOS/nixpkgs/blob/fc553c0bc5411478e2448a707f74369ae9351e96/pkgs/tools/misc/etcher/default.nix#L49y
+      sed -i "
+        s|/usr/bin/pkexec|/run/wrappers/bin/pkexec|
+        s|/bin/bash|${bash}/bin/bash|
+      " "$unpacked/sudo-prompt/index.js"
+      # delete original .asar file, as the new unpacked is now replacing it
+      rm -rf "$packed"
+      # PATCHING GLOBAL SEARCH
+      # ----------------------
+      chmod +x resources/app/node_modules/vscode-ripgrep/bin/rg
```

Comment:
That sounds like it should be part of a different PR?

Reply:
Or commit

Reply:
Well, this patch fixes them both, I'm not sure why is it necessary to split.

Reply:
Because they fix different issues. If we have to revert one of them, for example, we don't want to revert the other one. Or if we want to bisect to search for a fix or a bug, it is important that they are separate.